frontend/src/pages/auth/signin/SignIn.tsx

```diff
@@ -1,11 +1,52 @@
 import React from 'react';
-import {Titles} from "../../../constansts/titles";
+import {Box, Button} from "@mui/material";
+import "../../../common/styles/button.css";
+import {useAppDispatch} from "../../../stores/redux/hooks";
+import {signIn} from "../../../stores/redux/loggedUser/loggedUserSlice";
+import {getUserService} from "../../../services/user";
+import {Buttons} from "../../../common/constansts/buttons";
+import {logInService} from "../../../services/auth";
+import {User} from "../../../models/user";
+import {UserType} from "../../../types/UserType";
+import {ILoggedUser} from "../../../types/loggedUser";
 
 const SignIn: React.FC = () => {
+    const dispatch = useAppDispatch()
+
+    const decodeToken = (token: string) => {
```

Comment:
move this method to common/utils.ts
Add return type.
Type should be full token model